### PR TITLE
Code clean up part 2

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
 - Removes code duplication
 - Reuse sync function
 - Add more tests

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/master/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/master/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: pkg/deploy/role_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	done, err := SyncRoleToCluster(deployContext, "test", []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"test-1"},
+			Resources: []string{"test-1"},
+			Verbs:     []string{"test-1"},
+		},
+	})
+	if !done || err != nil {
+		t.Fatalf("Failed to sync crb: %v", err)
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/role_test.go:53]: Minor:

crb -> role

File: pkg/deploy/tls.go [hunk 2]
```diff
@@ -212,8 +207,10 @@ func GetEndpointTLSCrtChain(deployContext *DeployContext, endpointURL string) ([
 			var ingress *v1beta1.Ingress
 			for wait := true; wait; {
 				time.Sleep(time.Duration(1) * time.Second)
-				ingress, err = GetClusterIngress(ingressSpec.Name, ingressSpec.Namespace, deployContext.ClusterAPI.Client)
-				if err != nil {
+
+				ingress := &v1beta1.Ingress{}
+				exists, err := GetNamespacedObject(deployContext, ingressSpec.Name, ingress)
+				if !exists {
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/tls.go:213]: Maybe:
if !exists || err != nil {

File: pkg/deploy/tls.go [hunk 3]
```diff
@@ -330,14 +327,14 @@ func K8sHandleCheTLSSecrets(deployContext *DeployContext) (reconcile.Result, err
...
+		if !done {
+			return reconcile.Result{}, err
 		}
 
-		roleBiding, err := SyncRoleBindingToCluster(deployContext, CheTLSJobRoleBindingName, CheTLSJobServiceAccountName, CheTLSJobRoleName, "Role")
-		if roleBiding == nil {
-			return reconcile.Result{RequeueAfter: time.Second}, err
+		done, err = SyncRoleBindingToCluster(deployContext, CheTLSJobRoleBindingName, CheTLSJobServiceAccountName, CheTLSJobRoleName, "Role")
+		if !done {
+			return reconcile.Result{}, err
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/tls.go:337]: Maybe log error:
``` go
			if err != nil {
				logrus.Error(err)
			}
			return reconcile.Result{RequeueAfter: time.Second}, err
```

**Review by AndrienkoAleksandr — APPROVED** [review]

File: pkg/deploy/tls.go (hunk 3 above)
Comment (tolusha) [reply pkg/deploy/tls.go:337]: Error will be printed by che_controller.go
https://github.com/eclipse-che/che-operator/blob/master/pkg/controller/che/che_controller.go#L511-L519

**Pull request merged**